paddle/fluid/platform/device_tracer.cc

```diff
@@ -370,21 +487,61 @@ DeviceTracer *GetDeviceTracer() {
   return tracer;
 }
 
-void SetCurAnnotation(const std::string &anno) {
-  annotation_stack.push_back(anno);
-}
+void SetCurAnnotation(Event *event) { annotation_stack.push_back(event); }
 
 void ClearCurAnnotation() { annotation_stack.pop_back(); }
 
-std::string CurAnnotation() {
-  if (annotation_stack.empty()) return "";
+Event *CurAnnotation() {
+  if (annotation_stack.empty()) return nullptr;
   return annotation_stack.back();
 }
+std::string CurAnnotationName() {
+  if (annotation_stack.empty()) return "";
+  return annotation_stack.back()->name();
+}
 
 void SetCurBlock(int block_id) { block_id_stack.push_back(block_id); }
 
 void ClearCurBlock() { block_id_stack.pop_back(); }
 
 int BlockDepth() { return block_id_stack.size(); }
+
+uint32_t GetCurSystemThreadId() {
+  std::stringstream ss;
+  ss << std::this_thread::get_id();
+  uint32_t id = static_cast<uint32_t>(std::stoull(ss.str()));
+  return id;
+}
+
+int64_t GetCurSystemThreadIdFromPthread() {
+#ifdef __linux__
+  return static_cast<int64_t>(pthread_self());
+#endif
+  return 0;
+}
+
+void SetCurThreadId(int32_t id) {
+  /*
+    NOTE(liangdun): In order to map CUDA API thread id into paddle's thread id,
+    we have to get current thread id. However, I can not find other elegant and
+    portable way to get integer thread id. One way is get thread id from
+    pthread. But this method only works on Linux. Another way is using
+    stringstream, this method is more portable. These two methods are
+    equivalent In Linux.
+  */
+  auto gid = GetCurSystemThreadId();
+  auto gid_p = GetCurSystemThreadIdFromPthread();
```

Comment:
We should use `gid` here, and `GetCurSystemThreadIdFromPthread ` should be removed.

Reply:
removed